fpga_interchange/dedicated_interconnect.cc

```diff
@@ -52,6 +58,44 @@ struct WireNode
 // interconnect.
 constexpr int kMaxDepth = 6;
 
+static uint32_t get_num_pips(const Context *ctx, WireId wire, ExpansionDirection direction)
```

Comment:
I think this should also exclude pseudo-pips - there may be a few cases of dedicated interconnect where multiple pseudo-pips are downstream.

1 should be alright for basic Xilinx stuff but we might need to play with the pip count threshold to get good results on other arches.